pkg/koordlet/metricsadvisor/collectors/nodeinfo/node_info_collector_test.go

```diff
@@ -179,41 +176,297 @@ Node 1 HugePages_Surp:        0`
+							PageSize: koordletutil.Hugepage1Gkbyte,
+							NumPages: 20,
+						},
+					},
+				},
 			},
 		},
-		MemInfoMap: map[int32]*koordletutil.MemInfo{
-			0: testMemInfo0,
-			1: testMemInfo1,
+		{
+			name: "memory and 2M & 1G hugepage",
+			initfunc: func() {
+				numaMemInfoPath0 := system.GetNUMAMemInfoPath("node0")
+				helper.WriteFileContents(numaMemInfoPath0, numaMemInfoContentStr0)
+				numaMemInfoPath1 := system.GetNUMAMemInfoPath("node1")
+				helper.WriteFileContents(numaMemInfoPath1, numaMemInfoContentStr1)
+				numaHugePage1GPath0 := system.GetNUMAHugepagesNrPath("node0", "hugepages-1048576kB")
+				helper.WriteFileContents(numaHugePage1GPath0, "10")
+				numaHugePage2MPath0 := system.GetNUMAHugepagesNrPath("node0", "hugepages-2048kB")
+				helper.WriteFileContents(numaHugePage2MPath0, "20")
+
+				numaHugePage1GPath1 := system.GetNUMAHugepagesNrPath("node1", "hugepages-1048576kB")
+				helper.WriteFileContents(numaHugePage1GPath1, "10")
+				numaHugePage2MPath1 := system.GetNUMAHugepagesNrPath("node1", "hugepages-2048kB")
+				helper.WriteFileContents(numaHugePage2MPath1, "20")
+			},
+			expected: &koordletutil.NodeNUMAInfo{
+				NUMAInfos: []koordletutil.NUMAInfo{
+					{
+						NUMANodeID: 0,
+						MemInfo:    testMemInfo0,
+						HugePages: map[uint64]*koordletutil.HugePagesInfo{
+							koordletutil.Hugepage2Mkbyte: {
+								PageSize: koordletutil.Hugepage2Mkbyte,
+								NumPages: 20,
+							},
+							koordletutil.Hugepage1Gkbyte: {
+								PageSize: koordletutil.Hugepage1Gkbyte,
+								NumPages: 10,
+							},
+						},
+					},
+					{
+						NUMANodeID: 1,
+						MemInfo:    testMemInfo1,
+						HugePages: map[uint64]*koordletutil.HugePagesInfo{
+							koordletutil.Hugepage2Mkbyte: {
+								PageSize: koordletutil.Hugepage2Mkbyte,
+								NumPages: 20,
+							},
+							koordletutil.Hugepage1Gkbyte: {
+								PageSize: koordletutil.Hugepage1Gkbyte,
+								NumPages: 10,
+							},
+						},
+					},
+				},
+				MemInfoMap: map[int32]*koordletutil.MemInfo{
+					0: testMemInfo0,
+					1: testMemInfo1,
+				},
+				HugePagesMap: map[int32]map[uint64]*koordletutil.HugePagesInfo{
+					0: {
+						koordletutil.Hugepage2Mkbyte: {
+							PageSize: koordletutil.Hugepage2Mkbyte,
+							NumPages: 20,
+						},
+						koordletutil.Hugepage1Gkbyte: {
+							PageSize: koordletutil.Hugepage1Gkbyte,
+							NumPages: 10,
+						},
+					},
+					1: {
+						koordletutil.Hugepage2Mkbyte: {
+							PageSize: koordletutil.Hugepage2Mkbyte,
+							NumPages: 20,
+						},
+						koordletutil.Hugepage1Gkbyte: {
+							PageSize: koordletutil.Hugepage1Gkbyte,
+							NumPages: 10,
+						},
+					},
+				},
+			},
 		},
 	}
-	t.Run("test", func(t *testing.T) {
-		c := &nodeInfoCollector{
-			collectInterval: 60 * time.Second,
-			storage:         metricCache,
-			started:         atomic.NewBool(false),
-		}
+	for _, tt := range tts {
+		t.Run(tt.name, func(t *testing.T) {
+			defer helper.Cleanup()
+			tt.initfunc()
```

Comment:
How about new the helper inside the test case and pass it as an argument of the `initFunc()`?

Reply:
good idea